frame/transaction-storage/src/benchmarking.rs

```diff
@@ -0,0 +1,170 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Benchmarks for transaction-storage Pallet
+
+#![cfg(feature = "runtime-benchmarks")]
+
+use sp_std::*;
+use super::*;
+use sp_runtime::traits::{Zero, One, Bounded};
+use sp_transaction_storage_proof::TransactionStorageProof;
+use frame_system::{RawOrigin, Pallet as System, EventRecord};
+use frame_benchmarking::{benchmarks, whitelisted_caller, impl_benchmark_test_suite};
+use frame_support::{traits::{Currency, OnFinalize, OnInitialize}};
+
+use crate::Pallet as TransactionStorage;
+
+const MAX_EXTRINSICS: u32 = 100;
```

Comment:
Should be configurable in the pallet. Also I dont see where this is enforced.

Reply:
This is only used in benchmarks

Reply:
But it assumes some data limit about the storage?

If you can only handle 100 extrinsics in benchmarking, i would expect to see some limit to the number of times a user can call the storage pallet in a single block... right?